miio/integrations/ihcooker/ihcooker.py

```diff
@@ -0,0 +1,565 @@
+
+    @command(
+        click.argument("temperature", type=int),
+        click.argument("skip_confirmation", type=bool, default=False),
+        click.argument("minutes", type=int, default=60),
+        click.argument("power", type=int, default=DEFAULT_FIRE_LEVEL),
+        click.argument("menu_location", type=int, default=None),
+        default_output=format_output("Cooking with temperature requested."),
+    )
+    def start_temp(
+        self,
+        temperature,
+        minutes=60,
+        power=DEFAULT_FIRE_LEVEL,
+        skip_confirmation=False,
+        menu_location=None,
+    ):
+        """Start cooking at a fixed temperature and duration.
+
+        Temperature in celcius.
+
+        Please do not use skip_confirmation=True, as this is potentially unsafe.
+        """
+
+        profile = dict(
+            recipe_name="%d Degrees" % temperature,
+            menu_settings=dict(save_recipe=False, confirm_start=skip_confirmation),
+            duration_minutes=minutes,
+            menu_location=menu_location,
+            stages=[
+                dict(
+                    temp_target=temperature,
+                    minutes=minutes,
+                    mode=StageMode.TemperatureMode,
+                    power=power,
+                )
+            ],
+        )
+        profile = self._prepare_profile(profile)
+
+        if menu_location is not None:
+            self.set_menu(profile, menu_location, True)
+        else:
+            self.start(profile, skip_confirmation)
+
+    @command(
+        click.argument("power", type=int),
+        click.argument("skip_confirmation", type=bool),
+        click.argument("minutes", type=int),
+        default_output=format_output("Cooking with temperature requested."),
+    )
+    def start_fire(self, power, minutes=60, skip_confirmation=False):
+        """Start cooking at a fixed fire power and duration.
+
+        Fire: 0-99.
+
+        Please do not use skip_confirmation=True, as this is potentially unsafe.
+        """
+
+        if 0 < power > 100:
+            raise ValueError("power should be in range [0,99].")
+        profile = dict(
+            recipe_name="%d fire power" % power,
+            menu_settings=dict(save_recipe=False, confirm_start=skip_confirmation),
+            duration_minutes=minutes,
+            stages=[dict(power=power, minutes=minutes, mode=StageMode.FireMode)],
+        )
+        profile = self._prepare_profile(profile)
+        self.start(profile, skip_confirmation)
+
+    @command(default_output=format_output("Cooking stopped"))
+    def stop(self):
+        """Stop cooking."""
+        self.send("set_func", ["end"])
+
+    @command(default_output=format_output("Recipe deleted"))
+    def delete_recipe(self, location):
+        """Delete recipe at location [0,7]"""
+        if location >= 9 or location < 1:
+            raise IHCookerException("location %d must be in [1,8]." % location)
```

Comment:
`start_fire` is using `ValueError`, maybe consolidate this (and others if necessary) to raise that standard error on invalid values?

Reply:
Done!